src/components/organisms/SettingsDrawer/Settings.tsx

```diff
@@ -0,0 +1,370 @@
+  BrowseKubeconfigTooltip,
+  HelmPreviewModeTooltip,
+  KubeconfigPathTooltip,
+  KustomizeCommandTooltip,
+} from '@constants/tooltips';
+
+import {ProjectConfig} from '@models/appconfig';
+
+import {useAppDispatch, useAppSelector} from '@redux/hooks';
+import {updateShouldOptionalIgnoreUnsatisfiedRefs} from '@redux/reducers/main';
+import {isInClusterModeSelector} from '@redux/selectors';
+
+import FilePatternList from '@molecules/FilePatternList';
+
+// import {Themes, TextSizes, Languages} from '@models/appconfig';
+import {useFocus} from '@utils/hooks';
+
+import Colors from '@styles/Colors';
+
+const StyledDiv = styled.div`
+  margin-bottom: 20px;
+`;
+
+const StyledSpan = styled.span`
+  font-weight: 500;
+  font-size: 20px;
+  display: block;
+  margin-bottom: 6px;
+`;
+
+const StyledButton = styled(Button)`
+  margin-top: 10px;
+`;
+
+const HiddenInput = styled.input`
+  display: none;
+`;
+
+const StyledSelect = styled(Select)`
+  width: 100%;
+`;
+
+const StyledWarningOutlined = styled(
+  (props: {isKubeconfigPathValid: boolean; highlighted?: boolean; className: string}) => (
+    <WarningOutlined className={props.className} />
+  )
+)`
+  ${props =>
+    `color: ${
+      props.highlighted ? Colors.whitePure : !props.isKubeconfigPathValid ? Colors.redError : Colors.yellowWarning
+    }`};
+  ${props => `margin-left: ${props.highlighted ? '10px' : '5px'}`};
+  ${props => `padding-top: ${props.highlighted ? '5px' : '0px'}`};
+`;
+
+const StyledHeading = styled.h2`
+  font-size: 16px;
+  margin-bottom: 7px;
+`;
+
+type SettingsProps = {
+  config?: ProjectConfig | null;
+  onConfigChange?: Function;
+  isClusterPaneIconHighlighted?: boolean | null;
+  showLoadLastProjectOnStartup?: boolean | null;
+};
+
+export const Settings = ({
+  config,
+  onConfigChange,
+  isClusterPaneIconHighlighted,
+  showLoadLastProjectOnStartup,
+}: SettingsProps) => {
+  const dispatch = useAppDispatch();
+  const isSettingsOpened = Boolean(useAppSelector(state => state.ui.isSettingsOpen));
+
+  const resourceRefsProcessingOptions = useAppSelector(state => state.main.resourceRefsProcessingOptions);
+  const uiState = useAppSelector(state => state.ui);
+
+  const isInClusterMode = useAppSelector(isInClusterModeSelector);
+  const fileInput = useRef<HTMLInputElement>(null);
+  const [inputRef, focusInput] = useFocus<Input>();
+  const hasUserPerformedClickOnClusterIcon = useAppSelector(state => state.uiCoach.hasUserPerformedClickOnClusterIcon);
+  const wasRehydrated = useAppSelector(state => state.main.wasRehydrated);
+  const [isClusterActionDisabled, setIsClusterActionDisabled] = useState(
+    Boolean(!config?.kubeConfig?.path) || Boolean(!config?.kubeConfig?.isPathValid)
+  );
+  const [currentKubeConfig, setCurrentKubeConfig] = useState(config?.kubeConfig?.path);
+  const isEditingDisabled = uiState.isClusterDiffVisible || isInClusterMode;
+
+  const [localConfig, setLocalConfig] = useState<ProjectConfig | null | undefined>(config);
```

Comment:
wouldn't `projectConfig` be a better name? or `currentProjectConfig` ?
I'm just getting a bit confused about difference between `currentConfig`, `projectConfig` and `localConfig` across the files

Reply:
I will explain in a comment at bottom